presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java

```diff
@@ -71,15 +81,49 @@ protected DdlMatchResult verify(
             ChecksumQueryContext controlChecksumQueryContext,
             ChecksumQueryContext testChecksumQueryContext)
     {
-        Statement controlChecksumQuery = getChecksumQuery(control);
-        Statement testChecksumQuery = getChecksumQuery(test);
+        String controlChecksum = null;
+
+        if (!skipControl || saveSnapshot) {
+            Statement controlChecksumQuery = getChecksumQuery(control);
+            controlChecksumQueryContext.setChecksumQuery(formatSql(controlChecksumQuery));
+
+            controlChecksum = getOnlyElement(callAndConsume(
+                    () -> getHelperAction().execute(controlChecksumQuery, CONTROL_CHECKSUM, checksumConverter),
+                    stats -> stats.getQueryStats().map(QueryStats::getQueryId).ifPresent(controlChecksumQueryContext::setChecksumQueryId)).getResults());
+
+            if (saveSnapshot) {
+                try {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    String snapshot = objectMapper.writeValueAsString(controlChecksum);
```

Comment:
is this needed?. This is going to just take the string and give the same string back right?

Reply:
Changed. Thanks

Reply:
@ajaygeorge This is needed for string escaping special characters like quotation marks. My final tests caught the bug and I restored the above code.